cucumber/api/features/support/logs_helpers.rb

```diff
@@ -0,0 +1,43 @@
+# frozen_string_literal: true
```

Comment:
Here i have a different opinion, due to the following:
1. I do not want to change the log file, i think the tests should not change anything that may affects other tests which maybe relaying on the log file 
2. imo the bookmarks are reflected what happening in manual tests, which i want to check the logs in specific timeline to ignore background noises 
3. i build here infrastructure that will work also if we will change the tests to run in parallel
4. with this approach you can easily validate the logs multiple time in the same scenario without a background noises 

@jonahx tell what do think?

Reply:
I agree that we shouldn't change anything for other tests but other tests shouldn't rely on the logs that were created in other tests, so I think we can clean the log file

Reply:
I agree with Jenia's point and still think zeroing is the best approach here.  That said, I'm not absolutely opposed to your approach either.  If you do decide to go with it, here are some changes I'd consider.  Was easier to just make them in vim vs try to describe each one, but basically I removed a few checks that didn't make sense to me, and changed bookmark from a Hash to a single value, since that's all we're using ("the previous number of log lines").  Remember each scenario will be run in its own world.

This is untested but should get across all the ideas, and you can make the required changes to the step definitions which should be simple:

```ruby
# frozen_string_literal: true

# Utility methods for Logs steps
#
# TODO: Add here comments explaining the reasons we're doing this, 
# to parallelize tests, etc.
#
module LogsHelpers
  LOG_LOCATION = "/src/conjur-server/log/development.log"

  def num_log_lines
    File.new(LOG_LOCATION).readlines.size
  end

  # NOTE: Client code should explicitly call this to save current log size,
  # rather than making it a side affect of getting the current size
  def save_num_log_lines
    @saved_num_lines = num_log_lines
  end

  def num_matches_since_savepoint(msg)
    cur_num_lines = num_log_lines

    validate_savepoint_exists
    validate_log_didnt_shrink(cur_num_lines)

    start_line, end_line = @saved_num_lines, cur_num_lines
    (File.readlines(LOG_LOCATION)[start_line..end_line].grep(/#{msg}/i)).count
  end

  private

  def validate_savepoint_exists
    raise "No savepoint exists.  'save_num_log_lines' must be called before this method" unless @saved_num_lines
  end

  def validate_log_didnt_shrink(cur_num_lines)
    return if cur_num_lines >= @saved_num_lines
    raise "The log has fewer lines (#{cur_num_lines}) than it did before (#{@saved_num_lines})"
  end
end

World(LogsHelpers)
```

Reply:
Thanks, i changed according to your comment.
Note: i used the hash in order to set infrastructure for multiple bookmarks in the scenario but i actually didn't used it.
By the way, do you think we should add the logs validation to other features ?